src/util.c

```diff
@@ -82,12 +86,46 @@ nats_ParseInt64(const char *d, int dLen)
         if ((dec < ASCII_0) || (dec > ASCII_9))
             return -1;
 
-        n = (n * 10) + ((int64_t)dec - ASCII_0);
+        k = ((int64_t)dec - ASCII_0);
+
+        // Overflow
+        if (((n * 10) > (INT64_MAX - k)) || (n > (INT64_MAX / 10)))
+            return -1;
+
+        n = (n * 10) + k;
     }
 
     return n;
 }
 
+// nats_ParseInt expects decimal integer numbers.
```

Comment:
I would not add this function. Note that I had to remove use of strtol in the past due to locale and security issues.